Multisig/Features/Security/SecurityCenter.swift

```diff
@@ -94,8 +94,65 @@ class SecurityCenter {
         }
     }
 
-    private static func migrateFromKeychainStorageToSecurityEnclave() {
-        //TODO:
+    func migrateFromKeychainStorageToSecurityEnclave() {
+        let task = { [unowned self] passcode in
+            do {
+                let keys = try KeyInfo.all()
+                // Migrate private keys
+                keys.compactMap { try? $0.privateKey() }.forEach {
+                    SecurityCenter.shared.import(id: DataID(id: $0.id), data: $0.keyData) { result in
+                        // TODO: Rollback on error
+                    }
+                }
+
+                // Migrate delegate keys
+                keys.compactMap { try? $0.delegatePrivateKey() }.forEach {
+                    SecurityCenter.shared.import(id: DataID(id: $0.id), data: $0.keyData, protectionClass: .data) { result in
+                        // TODO: Rollback on error
+                    }
+                }
+
+                // Migrate to new lock method
+                if App.shared.auth.isPasscodeSetAndAvailable {
+                    var lockMethod = LockMethod.passcode
+                    if AppSettings.passcodeOptions.contains(.useBiometry) {
+                        lockMethod = .userPresence
+                    }
+
+                    AppSettings.securityLockMethod = lockMethod
+                    AppSettings.securityLockEnabled = true
+                    if AppSettings.passcodeOptions.contains(.useForConfirmation) {
+                        try changeStoreSettings(currentPlaintextPassword: nil,
+                                                newPlaintextPassword: passcode,
+                                                store: sensitiveStore)
+                    }
+
+                    if AppSettings.passcodeOptions.contains(.useForLogin) {
+                        try changeStoreSettings(currentPlaintextPassword: nil,
+                                                newPlaintextPassword: passcode,
+                                                store: dataStore)
+                    }
+                }
+
+                // Enable feature toggle
+                AppConfiguration.FeatureToggles.securityCenter = true
+            } catch {
+                Self.reset()
+                AppSettings.securityLockEnabled = false
+                AppSettings.securityLockMethod = .passcode
```

Comment:
what if there was no lock method set before the migration?

Reply:
so we are setting default values here, right?

Reply:
This work is still not final, so I'm not sure if I will change this part or not yet